compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -437,12 +437,12 @@ pub fn can_reference_module_id(
     // Note that if the target module is in a different crate from the current module then we will either
     // return true as the target module is public or return false as it is private without looking at the `CrateDefMap` in either case.
     let same_crate = target_module.krate == importing_crate;
-    let target_crate_def_map = &def_maps[&target_module.krate];
 
     match visibility {
         ItemVisibility::Public => true,
         ItemVisibility::PublicCrate => same_crate,
         ItemVisibility::Private => {
+            let target_crate_def_map = &def_maps[&target_module.krate];
```

Comment:
Moving this is unrelated to this PR, but accessing this hashmap is only needed here so I moved it now before I forget (I don't know if Rust optimizes this out, maybe not because it can't know if that call has side-effects, so less hash accesses, better compiler performance).